src/events/listenPullRequestOpenMessage.js

```diff
@@ -0,0 +1,197 @@
+  if (!text) {
+    return '';
+  }
+  return text
+    .toLowerCase()
+    .split(' ')
+    .map((word) => {
+      return word.charAt(0).toUpperCase() + word.slice(1);
+    })
+    .join(' ');
+}
+
+function toCamelCase(str) {
+  return str
+    .toLowerCase()
+    .replace(/[^a-zA-Z0-9]+(.)/g, (_, char) => char.toUpperCase());
+}
+
+function cleanStringArray(arr) {
+  return arr.filter((str) => str.trim() !== '');
+}
+
+function extractDataFromPRGitHubUrl(githubURL) {
+  if (!githubURL) {
+    return;
+  }
+
+  const [organization, repository, type, pullNumber] = new URL(
+    githubURL
+  ).pathname
+    .split('/')
+    .slice(1);
+
+  return {
+    organization,
+    repository,
+    type,
+    pullNumber,
+  };
+}
+
+function isPullRequestOpen(prTitle = '') {
+  return /pull request opened/i.test(prTitle);
+}
+
+async function fetchPullRequest(pullRequestMetadata) {
+  const owner = pullRequestMetadata.author.name;
+  const { repository: repo, pullNumber } = extractDataFromPRGitHubUrl(
+    pullRequestMetadata.url
+  );
+
+  if (!repo || !pullNumber) {
+    throw new Error(
+      translateLanguage('pullRequestOpen.errorExtractDataFromURL')
+    );
+  }
+
+  const url = `https://api.github.com/repos/${owner}/${repo}/pulls/${pullNumber}`;
+  const response = await fetch(url, { headers });
+
+  if (!response.ok) {
+    throw new Error(
+      translateLanguage('pullRequestOpen.errorFetchPR', {
+        status: response.status,
+      })
+    );
+  }
+
+  const data = await response.json();
+  return data;
+}
+
+function extractPRMetadata(description = '') {
+  if (typeof description !== 'string' || !description.trim()) {
+    return {};
+  }
+
+  let lastSectionKey = null;
+  const sectionMap = {};
+  const lines = cleanStringArray(description.split('\r\n'));
+
+  for (const line of lines) {
+    if (line.includes('## ')) {
+      const sectionTitle = line.substring(3).trim().toLowerCase();
+      const sectionKey = toCamelCase(sectionTitle);
+      sectionMap[sectionKey] = {
+        title: sectionTitle,
+        description: '',
+      };
+      lastSectionKey = sectionKey;
+    } else if (lastSectionKey) {
+      sectionMap[lastSectionKey].description += `${line}\n`;
+    }
+  }
+
+  // clean data and extract url image
+  Object.entries(sectionMap).forEach(([sectionKey, section]) => {
+    if (!sectionKey.includes('screenshot')) {
+      return (section.description = section.description.trim());
+    }
+
+    const match = section.description.match(/!\[.*?\]\((.*?)\)/);
+    section.description = match ? match[1] : null;
+  });
+
+  return sectionMap;
+}
+
+function generateOverview(prRestMetadata) {
+  const prMetadataKeys = Object.keys(prRestMetadata);
+  const prOverview = prMetadataKeys.reduce((overview, metadataKey, index) => {
+    return `${
+      overview
+    }## ${capitalizeText(prRestMetadata[metadataKey].title)}\n${prRestMetadata[metadataKey].description}${prMetadataKeys.length - 1 === index ? '' : '\n'}`;
```

Comment:
The spaces inside `${}` looks a bit weird, as I am not sure if they are actually modifying the string?, I think those don't affect the string

```suggestion
    return `${overview}## ${capitalizeText(prRestMetadata[metadataKey].title)}\n${prRestMetadata[metadataKey].description}${prMetadataKeys.length - 1 === index ? '' : '\n'}`;
```

please confirm

Reply:
I think it was prettier